src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java

```diff
@@ -140,22 +145,19 @@ public JurorDetailsCommonResponseDto(JurorPool jurorPool,
         this.excusalRejected = juror.getExcusalRejected();
         this.excusalCode = juror.getExcusalCode();
         this.noDeferrals = juror.getNoDefPos();
-
         this.poolNumber = jurorPool.getPoolNumber();
         this.startDate = jurorPool.getReturnDate();
         this.deferralDate = jurorPool.getDeferralDate();
         this.deferredTo = jurorPool.getDeferralDate() != null ? jurorPool.getDeferralDate() : null;
+        this.deferralCode = jurorPool.getDeferralCode();
         this.courtName = jurorPool.getCourt().getLocCourtName();
 
         if (this.excusalCode != null) {
-            List<ResponseExcusalController.ExcusalCodeDto> excusalCodeDtoList =
-                responseExcusalService.getExcusalReasons();
-
-            Optional<ResponseExcusalController.ExcusalCodeDto> excusalCodeReasonOpt = excusalCodeDtoList.stream()
-                .filter(code -> code.getExcusalCode().equals(this.excusalCode)).findFirst();
-
-            this.excusalDescription =
-                excusalCodeReasonOpt.map(ResponseExcusalController.ExcusalCodeDto::getDescription).orElse(null);
+            this.excusalDescription =  ExcusalCodeEnum.valueOf(this.excusalCode).getDescription();
```

Comment:
Do we need validation here to ensure excusal code exists within the enum?

Reply:
updated to throw internal server error if enum value is not found in DB